### ISSUE TITLE
cannot use exported packages on `x64-linux`, using manifest mode

### ISSUE THREAD

**Issue by klochowicz** [microsoft/vcpkg#36095]
**Describe the bug**

First of all: without an export, everything builds & runs correctly.

After exporting packages with `vcpkg export`, the triplet is not parsed correctly.
I can see `x64-linux` directories in the export, but no triplet is found.

**Environment**
- OS: Linux (Ubuntu/WSL)
- Compiler: GCC 11.4.0

**To Reproduce**
1. ./vcpkg export --target-dir=exported --zip
2. Use the exported CMake toolchain file in the build
3. Try to build the project

**Expected behavior**
Exported packages should be usable. Triplet `x64-linux` should  be found correctly.

**Failure logs**

Relevant part from running CMake:

```
-- Running vcpkg install
error: Invalid triplet: x64-linux
Built-in Triplets:
Community Triplets:
See https://learn.microsoft.com/vcpkg/users/triplets for more information.
-- Running vcpkg install - failed
CMake Error at export/vcpkg-export-20240109-123617/scripts/buildsystems/vcpkg.cmake:899 (message):
  vcpkg install failed.  See logs for more information:
  /home/mariusz/git/monorepo/wamsoft/build/Debug/vcpkg-manifest-install.log
Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/CMakeDetermineSystem.cmake:124 (include)
  CMakeLists.txt:74 (project)
```

**Comment by klochowicz** [comment]
also: exactly the same happens when using `x64-mingw-static`.

**Comment by klochowicz** [comment]
additional logs with `--debug` during `vcpkg export`:

```
[DEBUG] To include the environment variables in debug output, pass --debug-env
[DEBUG] Disabling metrics because vcpkg.disable-metrics exists
[DEBUG] Trying to load bundleconfig from /home/mariusz/git/vcpkg/vcpkg-bundle.json
[DEBUG] Failed to open: /home/mariusz/git/vcpkg/vcpkg-bundle.json
[DEBUG] Bundle config: readonly=false, usegitregistry=false, embeddedsha=nullopt, deployment=Git, vsversion=nullopt
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Feature flag 'dependencygraph' unset
[DEBUG] Using scripts-root: /home/mariusz/git/vcpkg/scripts
[DEBUG] Using builtin-ports: /home/mariusz/git/vcpkg/ports
[DEBUG] Using installed-root: /home/mariusz/git/monorepo/wamsoft/vcpkg_installed
[DEBUG] Using buildtrees-root: /home/mariusz/git/vcpkg/buildtrees
[DEBUG] Using packages-root: /home/mariusz/git/vcpkg/packages
[DEBUG] Using manifest-root: /home/mariusz/git/monorepo/wamsoft
[DEBUG] Using vcpkg-root: /home/mariusz/git/vcpkg
[DEBUG] Using scripts-root: /home/mariusz/git/vcpkg/scripts
[DEBUG] Using builtin-registry: /home/mariusz/git/vcpkg/versions
[DEBUG] Using downloads-root: /home/mariusz/git/vcpkg/downloads
```

**Comment by klochowicz** [comment]
also, because it might be relevant: my `vcpkg.json` looks something like this one (with built-in baseline)

```
{
  "name": "project",
  "version": "1.0.0",
  "dependencies": ["zlib", "fmt"],
  "builtin-baseline":"9fd3bd594f41afb8747e20f6ac9619f26f333cbe"
}
```

**Comment by autoantwort** [comment]
Delete you manifest file and try again

**Comment by klochowicz** [comment]
this indeed is a workaround for this issue, thank you @autoantwort ! 

is the the final UX for this feature though? `vcpkg.json` as this contains the blueprint of my dependencies, which I should version in git (this is how I can create an export in the first place...). 

for the time being, I can probably rename it back and forth, but it's not ideal....

**Comment by eclipse0922** [comment]
another workaround 
`set(VCPKG_MANIFEST_DIR "") # manually disable manifest mode`

**Comment by dg0yt** [comment]
> another workaround in CMakeLists.txt or manual arguments
> `set(VCPKG_MANIFEST_DIR "") # manually disable manifest mode`

Why not use the official `VCPKG_MANIFEST_MODE` options?

**Comment by eclipse0922** [comment]
@dg0yt thank you for sharing
here is offical doc
https://learn.microsoft.com/en-us/vcpkg/users/buildsystems/cmake-integration#vcpkg_manifest_mode

**Comment by marekp0** [comment]
Just ran into this issue. For anyone else stumbling onto this, here's some summary/clarification of the above:

The cleanest workaround is to just set `VCPKG_MANIFEST_MODE` to `OFF` in your CMakeLists.txt, CMakePresets.json, or manually (`-DVCPKG_MANIFEST_MODE=off`) when trying to use the exported packages. I don't think hard-coding it in CMakeLists.txt is a great idea, because it then prevents vcpkg from installing packages from your manifest when you're not using exported packages.

Figuring out the right thing to google for this was quite the adventure...

**Comment by kya8** [comment]
I just ran into this. I suppose manifest mode does not make sense with using the exported tree?

Maybe we should clarify this in the docs, that when using an exported tree, manifest mode should be disabled. Or should this be handled automatically in the exported toolchain file?

**Comment by iDommel** [comment]
I have the same issue on x64-windows, in manifest mode, trying to use an exported version of packages.
Adding "VCPKG_MANIFEST_MODE": "OFF" to the cmake preset still results in the same error message of 
error: Invalid triplet: x64-windows

I assume that by default it still tries to detect what triplet to use before doing anything else.
Did anyone else get this issue on windows ?

**Comment by dg0yt** [comment]
@iDommel Do you configure the same host and target triplets for *using* the export as for *creating* the export?  (I run this setup in GHA, so it works. But I know there are dark areas.)

**Comment by dg0yt** [comment]
@kya8 @klochowicz I don't think it is technically supported to combine exported packages with a manifest mode install tree. Basically you pick a single vcpkg install tree with the vcpkg toolchain file. The toolchain file will provide certain services - such as vcpkg cmake wrappers - only from the selected tree.
And then there is all the normal trouble with mixing multiple dependency providers, including system packages.

**Comment by iDommel** [comment]
> [@iDommel](https://github.com/iDommel) Do you configure the same host and target triplets for _using_ the export as for _creating_ the export? (I run this setup in GHA, so it works. But I know there are dark areas.)

I do, I've tried x64-windows and x64-windows-release (both official triplets) what I noticed is that in the exported folder while there is a cmake toolchain file and a vcpkg executable there is no triplets folder. So if you set your toolchain file to be that of the exported folder instead of locally installed vcpkg, it looses context for which triplets are available.

Maybe you are supposed to chainload that toolchain with the main vcpkg toolchain from your local install ? I haven't touched chainloading much.
